persistence/schema/gov.go

```diff
@@ -1,442 +1,145 @@
-		"proposer_percentage_of_fees_owner",
-		"missed_blocks_burn_percentage_owner",
-		"double_sign_burn_percentage_owner",
-		"message_double_sign_fee_owner",
-		"message_send_fee_owner",
-		"message_stake_fisherman_fee_owner",
-		"message_edit_stake_fisherman_fee_owner",
-		"message_unstake_fisherman_fee_owner",
-		"message_pause_fisherman_fee_owner",
-		"message_unpause_fisherman_fee_owner",
-		"message_fisherman_pause_service_node_fee_owner",
-		"message_test_score_fee_owner",
-		"message_prove_test_score_fee_owner",
-		"message_stake_app_fee_owner",
-		"message_edit_stake_app_fee_owner",
-		"message_unstake_app_fee_owner",
-		"message_pause_app_fee_owner",
-		"message_unpause_app_fee_owner",
-		"message_stake_validator_fee_owner",
-		"message_edit_stake_validator_fee_owner",
-		"message_unstake_validator_fee_owner",
-		"message_pause_validator_fee_owner",
-		"message_unpause_validator_fee_owner",
-		"message_stake_service_node_fee_owner",
-		"message_edit_stake_service_node_fee_owner",
-		"message_unstake_service_node_fee_owner",
-		"message_pause_service_node_fee_owner",
-		"message_unpause_service_node_fee_owner",
-		"message_change_parameter_fee_owner",
-		"end_height",
-	}
+	govParamMetadataMap  map[string]govParamMetadata
+	govParamMetadataKeys []string
+
+	ValTypeEnumTypes = fmt.Sprintf(`(
+		'%s',
+		'%s',
+		'%s'
+		)`,
+		ValTypeString,
+		ValTypeBigInt,
+		ValTypeSmallInt,
+	)
 )
 
 func InsertParams(params *genesis.Params) string {
-	return fmt.Sprintf(`INSERT INTO %s VALUES(%d, '%s', %d, %d, %d, %d, %d, %d,
-						'%s',%d,%d,%d,%d,%d,
-						'%s',%d,%d,%d,%d,
-						'%s',%d,%d,%d,%d,
-						%d,%d,%d,%d,
-						'%s','%s','%s','%s','%s','%s','%s','%s','%s','%s',
-						'%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s',
-						'%s','%s','%s','%s','%s','%s','%s','%s','%s',
-						'%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s',
-						'%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s',%d)`,
-		ParamsTableName,
-		params.BlocksPerSession,
-		params.AppMinimumStake,
-		params.AppMaxChains,
-		params.AppBaselineStakeRate,
-		params.AppStakingAdjustment,
-		params.AppUnstakingBlocks,
-		params.AppMinimumPauseBlocks,
-		params.AppMaxPauseBlocks,
-
-		params.ServiceNodeMinimumStake,
-		params.ServiceNodeMaxChains,
-		params.ServiceNodeUnstakingBlocks,
-		params.ServiceNodeMinimumPauseBlocks,
-		params.ServiceNodeMaxPauseBlocks,
-		params.ServiceNodesPerSession,
-
-		params.FishermanMinimumStake,
-		params.FishermanMaxChains,
-		params.FishermanUnstakingBlocks,
-		params.FishermanMinimumPauseBlocks,
-		params.FishermanMaxPauseBlocks,
-
-		params.ValidatorMinimumStake,
-		params.ValidatorUnstakingBlocks,
-		params.ValidatorMinimumPauseBlocks,
-		params.ValidatorMaxPauseBlocks,
-		params.ValidatorMaximumMissedBlocks,
+	val := reflect.ValueOf(params)
```

Comment:
NIT: Please add a 'WARNING: reflections in prod' comment.

It's always a bit scary but I like this approach much more